__tests__/commands/run-integration.js

```diff
@@ -0,0 +1,57 @@
+/* @flow */
+
+import makeTemp from '../_temp.js';
+import * as fs from '../../src/util/fs.js';
+import {run as buildRun} from './_helpers.js';
+import {BufferReporter} from '../../src/reporters/index.js';
+import {run} from '../../src/cli/commands/run.js';
+
+const path = require('path');
+
+const fixturesLoc = path.join(__dirname, '..', 'fixtures', 'run');
+const runRun = buildRun.bind(null, BufferReporter, fixturesLoc, (args, flags, config, reporter): Promise<void> => {
+  return run(config, reporter, flags, args);
+});
+
+jasmine.DEFAULT_TIMEOUT_INTERVAL = 30000;
+
+test('fully quote and escape extra args to run', async () => {
+  const trickyStrings = ['$X', '%X%', '^', '!', '\\', '>', '<', '|', '&', "'", '"', '`', '  '];
+  const cases = [];
+  for (const a of trickyStrings) {
+    for (const b of trickyStrings) {
+      cases.push(a + b + b + b);
+      if (a !== b) {
+        cases.push(a + a + b + b);
+        cases.push(a + b + a + b);
+        cases.push(a + b + b + a);
+        cases.push(a + a + a + b);
+        cases.push(a + a + b + a);
+        cases.push(a + b + a + a);
+      }
+    }
+  }
+  cases.sort(() => Math.random() - 0.5);
```

Comment:
Any non-determinism in tests would create doubt when running things on the CI: did it fail for reals or was that a random blip? Did I catch a real edge case thanks to randomness, or the CI had a bad network connection? So let's have this test all possible combinations instead of random sampling for each test run or agree to ahve the same random sample for each run.